synapse/storage/roommember.py

```diff
@@ -501,40 +501,34 @@ def _get_joined_users_from_context(self, room_id, state_group, current_state_ids
 
         defer.returnValue(users_in_room)
 
-    def is_host_joined(self, room_id, host, state_group, state_ids):
-        if not state_group:
-            # If state_group is None it means it has yet to be assigned a
-            # state group, i.e. we need to make sure that calls with a state_group
-            # of None don't hit previous cached calls with a None state_group.
-            # To do this we set the state_group to a new object as object() != object()
-            state_group = object()
+    @defer.inlineCallbacks
+    def is_host_joined(self, room_id, host):
+        if '%' in host or '_' in host:
+            raise Exception("Invalid host name")
+
+        sql = """
+            SELECT state_key FROM current_state_events
+            INNER JOIN room_memberships USING (room_id, event_id)
+            WHERE membership = 'join' AND room_id = ? AND state_key LIKE ?
+            LIMIT 1
```

Comment:
> I think you might want to remove the "room_id" from the join since we want it to force it to join using the room_memberships_event_id_key index.

It's hardly going to choose to join on room_id over event_id, but interestingly removing the room_id clause causes it to use an index rather than bitmap scan for me, for whatever reason.